src/cure.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// cure.sol -- Debt Rectifier contract
+
+// Copyright (C) 2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+
+interface SourceLike {
+    function cure() external view returns (uint256);
+}
+
+contract Cure {
+    mapping (address => uint256) public wards;
+    uint256 public live;
+    address[] public sources;
```

Comment:
Yeah let's keep it custom at least in this case.

Reply:
+1 to @gbalabasquer here, let's not profane `dss` with OZ code (or any other external dependency for that matter).

Reply:
Just to be clear I suggest copying the code instead of adding some external dependancy such as [here](https://github.com/makerdao/dss-wormhole/blob/master/src/utils/EnumerableSet.sol). In this case I can appreciate the argument against. I'm fine with custom code. Will review it carefully.